File: packages/tool-task/src/core/utils/prompt/prompt.models.ts
import type { _PromptModel, _PromptParamsModel } from '@tool/task/core/utils/prompt/_prompt.models';
import type { PROMPT_TYPE } from '@tool/task/core/utils/prompt/prompt.constants';

export type PromptParamsModel<TParams extends Array<PromptArgsModel>> = _PromptParamsModel<TParams>;

export type PromptModel<TParams extends Array<PromptArgsModel>> = _PromptModel<TParams>;

export type PromptTypeModel = `${PROMPT_TYPE}`;

export type PromptArgsModel = {
  isOptional?: boolean;
  key: string;
  message?: string;
  options?: Array<string>;
} & (
  | { basePath?: string; defaultValue?: string; type?: PROMPT_TYPE.DIRECTORY }
  | { basePath?: never; defaultValue?: Array<string>; type?: PROMPT_TYPE.CHECKBOX }
  | { basePath?: never; defaultValue?: string; type?: PromptTypeModel }
);

File: packages/lib-shared-py/src/lib_shared/core/utils/prompt/prompt_models.py
from typing import Protocol


class _PromptModel(Protocol):
    async def __call__(
        self,
        key: str,
        message: str | None = None,
        options: list[str] | None = None,
        is_multiple: bool = False,
        default_value: str | None = None,
    ) -> str | list[str]: ...


PromptModel = _PromptModel
